### PR TITLE
Introduced EditorConfig#initialData. Made config param optional

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced `EditorConfig#initialData`.

---

### Additional information

Part of https://github.com/ckeditor/ckeditor5/issues/1619.

Also, I've made `config` parameter in `Editor.create()` optional. There is no need for it and we also have an example in docs without using it.

### CODE REVIEW

**Comment by pjasiun** [comment]
The documentation of the `create` method is messy. It was messy before the change and we should fix it. It mixes 3 ways of data initialization with building from source or using builds.

I believed it should focus on data initialization, show the difference between various ways of initing data, explain when each way should be used. After that, in the end, it should tell the developer that all samples above are based on the editor build and what should need to do to create the editor from the source, with the proper link to the documentation about building the editor from the source.

**Comment by scofalik** [comment]
I rewrote docs.

File: src/classiceditor.js [hunk 1]
```diff
@@ -105,78 +108,81 @@ export default class ClassicEditor extends Editor {
 	 *				console.error( err.stack );
 	 *			} );
 	 *
-	 * Creating an instance when using CKEditor from source (make sure to specify the list of plugins to load and the toolbar):
+	 * This is the most convenient and common way to initialize the editor. The element's content will be used as the editor data.
```
Comment (pjasiun) [inline src/classiceditor.js]:
```suggestion
	 * The element's content will be used as the editor data.
```

This is not necessarily the most convenient way to initialize the editor. Also, this sentence does not help much. It is just a statistics ;)

File: src/classiceditor.js [hunk 2]
```diff
@@ -105,78 +108,81 @@ export default class ClassicEditor extends Editor {
...
+	 * See also the {@link module:core/editor/editorconfig~EditorConfig editor configuration documentation} to learn more about
+	 * customizing plugins, toolbar and other.
+	 *
 	 * @param {HTMLElement|String} sourceElementOrData The DOM element that will be the source for the created editor
 	 * or the editor's initial data.
 	 *
-	 * If a source element is passed, then its contents will be automatically
-	 * {@link module:editor-classic/classiceditor~ClassicEditor#setData loaded} to the editor on startup
+	 * If a DOM element is passed, its content will be automatically
+	 * {@link module:editor-classic/classiceditor~ClassicEditor#setData loaded} to the editor upon initialization
```
Comment (pjasiun) [inline src/classiceditor.js]: It is a wrong link. `setData` will not be used to load initial data. `data.init` will be used.

File: src/classiceditor.js (hunk 1 above)
Comment (scofalik) [reply src/classiceditor.js]: We give somebody two (and a half) ways to init editor. I wanted to write which is more common, so probably better for them.

But yeah it is not necessary.

File: src/classiceditor.js [hunk 3]
```diff
@@ -105,78 +108,81 @@ export default class ClassicEditor extends Editor {
...
+	 * See also the {@link module:core/editor/editorconfig~EditorConfig editor configuration documentation} to learn more about
+	 * customizing plugins, toolbar and other.
+	 *
 	 * @param {HTMLElement|String} sourceElementOrData The DOM element that will be the source for the created editor
 	 * or the editor's initial data.
 	 *
-	 * If a source element is passed, then its contents will be automatically
-	 * {@link module:editor-classic/classiceditor~ClassicEditor#setData loaded} to the editor on startup
+	 * If a DOM element is passed, its content will be automatically
+	 * {@link module:editor-classic/classiceditor~ClassicEditor#initData loaded} to the editor upon initialization
```
Comment (pjasiun) [inline src/classiceditor.js]: Do we have `ClassicEditor#initData` method?

Comment (pjasiun) [reply src/classiceditor.js]: I think you can just remove this link. It is not that important that `data#init` is used internally.

**Pull request merged**